app.js

```diff
@@ -1,27 +1,63 @@
-console.log('js file works')
+console.log("js file works");
+
+// const toggleComplete = () => {
+//   // get all todos
+//   let allToDos = document.querySelectorAll(".individualToDo");
+//   // loop through todos and addEventListener
+//   console.log(allToDos);
+//   allToDos.forEach((singleToDo) => {
+//     singleToDo.addEventListener("click", function () {
+//       singleToDo.classList.toggle("strikeout");
+//     });
+//   });
+// };
+// toggleComplete();
+
+// mark complete
+const toggleComplete = () => {
+  // console.log("it toggles");
+  // get parent node
+  let completed = document.querySelector("ul");
+  // add event listener
+  completed.addEventListener("click", (todo) => {
+    // target event listener and toggle class
+    todo.target.classList.toggle("strikeout");
```

Comment:
I'm glad you hit on this. This works great because the `target` of the click event is currently always the `<li>` that was clicked. Once you have some more elements in there, `target` might be other things, we can cross that bridge when we come to it. 

It's still a totally fine approach, to have a click handler on a parent, and after it is clicked, check what element was the `target` of the click, then do the right thing. Some people even do this on the `body` element, so the browser has only a single click handler for the entire page, and when the something is clicked, the mega-click-handler says "oh what was clicked? The Add Item button? Okay cool I'll add the item".

In react etc it's more normal to use `onClick` on specific elements just cause it's a bit easier to think that way when you have lots of components, but it doesn't actually matter. And where elements will be added an removed by the user, what you've done here will work for the new elements just as well as the old.

Reply:
I'd love to talk about this more after MVP is completed (or if it comes up along the way).